tests/unit/utils/test_utils.py

```diff
@@ -126,6 +127,32 @@ def test_makedirs(repo_dir):
     assert os.path.isdir(path_info.fspath)
 
 
+@pytest.mark.parametrize(
+    "parent",
+    [
+        (os.path.join("path", "to", "")),
+        (os.path.join("path", "to")),
+        (os.path.join("path", "")),
+        (os.path.join("path")),
+    ],
```

Comment:
I think all those test could be done with one, where parametrize child, path and expected result. But I guess it is a matter of taste.

Reply:
Parametrizing for one line test is just a boiler plate.

Reply:
I tried that one but I found parameterizing `expected_result` as well too confusing for myself (and, too much repetition).

Reply:
@skshetry I am ok with this as it is

Reply:
I also prefer having multiple asserts, but I think, this is easier to follow and understand.